src/compiler/factory.ts

```diff
@@ -3492,7 +3492,8 @@ namespace ts {
 
     export function parenthesizeConciseBody(body: ConciseBody): ConciseBody {
         const emittedBody = skipPartiallyEmittedExpressions(body);
-        if (emittedBody.kind === SyntaxKind.ObjectLiteralExpression) {
+        const leftMostExpression = isExpression(emittedBody) ? getLeftmostExpression(emittedBody) : undefined;

```

Comment:
I think this could just be:

```ts
if (!isBlock(body) && getLeftmostExpression(body).kind === SyntaxKind.ObjectLiteralExpresion)
```

I suggest using `!isBlock(body)` here as there are fewer comparisons necessary than a full `isExpression` test.